modules/auxiliary/scanner/nfs/nfsmount.rb

```diff
@@ -10,74 +10,101 @@ class MetasploitModule < Msf::Auxiliary
 
   def initialize
     super(
-      'Name'          => 'NFS Mount Scanner',
-      'Description'   => %q{
+      'Name' => 'NFS Mount Scanner',
+      'Description' => %q{
         This module scans NFS mounts and their permissions.
       },
-      'Author'	       => ['<tebo[at]attackresearch.com>'],
-      'References'     =>
-        [
-          ['CVE', '1999-0170'],
-          ['URL',	'https://www.ietf.org/rfc/rfc1094.txt']
-        ],
+      'Author'	=> ['<tebo[at]attackresearch.com>'],
+      'References' => [
+        ['CVE', '1999-0170'],
+        ['URL',	'https://www.ietf.org/rfc/rfc1094.txt']
+      ],
       'License'	=> MSF_LICENSE
     )
 
     register_options([
-      OptEnum.new('PROTOCOL', [ true, 'The protocol to use', 'udp', ['udp', 'tcp']])
+      OptEnum.new('PROTOCOL', [ true, 'The protocol to use', 'udp', ['udp', 'tcp']]),
+      OptAddressLocal.new('LHOST', [false, 'IP to match shares against', Rex::Socket.source_address]),
+      OptString.new('HOSTNAME', [false, 'Hostname to match shares against', ''])
     ])
 
+    register_advanced_options(
+      [
+        OptBool.new('Mountable', [false, 'Determine if an export is mountable', true]),
+      ]
+    )
+  end
+
+  def can_mount?(locations)
+    # attempts to validate if we'll be able to open it or not based on:
+    # 1. its a wildcard, thus we can open it
+    # 2. hostname isn't blank and its in the list
+    # 3. our IP is explicitly listed
+    # 4. theres a CIDR notation that we're included in.
+    return true unless datastore['Mountable']
+    return true if locations.include? '*' ||
+                                      (locations.include?(datastore['HOSTNAME']) && !datastore['HOSTNAME'].blank?) ||
+                                      locations.include?(datastore['LHOST'])
```

Comment:
Personal preference. These could be 3 separate return statements.

```suggestion
    return true if locations.include?('*')
    return true if locations.include?(datastore['LHOST']
    return true if (locations.include?(datastore['HOSTNAME']) && !datastore['HOSTNAME'].blank?)
```

Also, `LHOST` is an optional module option, which could be a blank string or an invalid IP address. A `.blank?` check would be reasonable here.

```suggestion
    return true if locations.include?('*')
    return true if (locations.include?(datastore['LHOST'] && !datastore['LHOST'].blank?)
    return true if (locations.include?(datastore['HOSTNAME']) && !datastore['HOSTNAME'].blank?)
```

Reply:
should switch the order of the `.blank?` and the actual check in the if statement to to speed up if the options aren't set.